app/controllers/chargeback_controller.rb

```diff
@@ -713,21 +713,14 @@ def get_tags_all(category)
 
   def get_cis_all
     @edit[:cb_assign][:cis] = {}
-    if @edit[:new][:cbshow_typ] == "enterprise"
-      e_id = MiqEnterprise.first
-      @edit[:cb_assign][:cis]["#{e_id.id}"] = "Enterprise"
-    elsif @edit[:new][:cbshow_typ] == "storage"
-      Storage.all.each do |s|
-        @edit[:cb_assign][:cis][s.id] = s.name
-      end
-    elsif @edit[:new][:cbshow_typ] == "ext_management_system"
-      ExtManagementSystem.all.each do |ms|
-        @edit[:cb_assign][:cis][ms.id] = ms.name
-      end
-    elsif @edit[:new][:cbshow_typ] == "ems_cluster"
-      EmsCluster.all.each do |cl|
-        @edit[:cb_assign][:cis][cl.id] = cl.name
+    classtype =
+      if @edit[:new][:cbshow_typ] == "enterprise"
+        MiqEnterprise
+      else
+        @edit[:new][:cbshow_typ].classify.constantize
```

Comment:
@martinpovolny should we put a whitelist of allowed values we will constantize here? I did that with https://github.com/ManageIQ/manageiq/pull/7251 and don't really like it but these deep hashes that allow anything in them forces us to have these big conditionals or a whitelist.